openjdk/build.gradle

```diff
@@ -153,19 +156,17 @@ model {
 
     tasks {
         $.binaries.withType(SharedLibraryBinarySpec).each {
-            def archName = it.targetPlatform.architecture.name.replaceAll('-', '_')
             def source = it.sharedLibraryFile
 
             // Copies the native library to a resource location that will be included in the jar.
-            task "copyNativeLib${archName}"(type: Copy) {
+            task "copySharedNativeLib${osdetector.arch}"(type: Copy) {
                 from source
-                rename '(.+)(\\.[^\\.]+)', "\$1-$archName\$2"
                 // This location will automatically be included in the jar.
                 into "build/resources/main/META-INF/native"
             }
 
             // Make sure we build and copy the native library to the output directory.
-            compileJava.dependsOn "copyNativeLib${archName}"
+            compileJava.dependsOn "copySharedNativeLib${osdetector.arch}"
```

Comment:
This isn't going to work, because `osdetector` is detecting the arch of the current machine and not the target platform for the build (e.g., machine is 64-bit but building a 32-bit library).